istioctl/cmd/vm-bootstrap.go

```diff
@@ -0,0 +1,630 @@
+			}
+			if dumpDir == "" {
+				err := deriveSSHMethod()
+				if err != nil {
+					return err
+				}
+			}
+
+			return nil
+		},
+		RunE: func(c *cobra.Command, args []string) error {
+			var configClient model.ConfigStore
+			var err error
+
+			if configClient, err = clientFactory(); err != nil {
+				return err
+			}
+
+			var entries []model.Config
+			var chosenNS string
+			if all {
+				entries, chosenNS, err = fetchAllWorkloadEntries(configClient)
+			} else {
+				entries, chosenNS, err = fetchSingleWorkloadEntry(args[0], configClient)
+			}
+			if err != nil {
+				return err
+			}
+
+			kubeClient, err := interfaceFactory(kubeconfig)
+			if err != nil {
+				return err
+			}
+
+			certs, err := getCertificate(kubeClient)
+			if err != nil {
+				return err
+			}
+
+			addresses, err := getCertificatesForEachAddress(entries, chosenNS, certs)
+			if err != nil {
+				return err
+			}
+
+			if dumpDir != "" {
+				err = dumpCertificates(dumpDir, addresses)
+				if err != nil {
+					return err
+				}
+			} else {
+				err = copyCertificates(kubeClient, addresses)
+				if err != nil {
+					return err
+				}
+			}
+
+			return nil
+		},
+	}
+
+	vmBSCommand.PersistentFlags().BoolVarP(&all, "all", "a", false,
+		"attempt to bootstrap all workload entries")
+	vmBSCommand.PersistentFlags().StringVarP(&dumpDir, "dump-directory", "d", "",
+		"dump the certificates locally in the provided directory instead of uploading remotely.")
+	vmBSCommand.PersistentFlags().DurationVar(&certDuration, "duration", 365*24*time.Hour,
+		"duration the certificates generated are valid for.")
+	vmBSCommand.PersistentFlags().StringVar(&istioProxyImage, "istio-image", "istio/proxyv2:latest",
+		"the istio proxy image to start up when starting istio")
+	vmBSCommand.PersistentFlags().BoolVar(&mutualTLS, "mutual-tls", false,
+		"Whether or not to enable mutual TLS if starting istio-proxy.")
+	vmBSCommand.PersistentFlags().StringVarP(&workloadNamespace, "namespace", "n", "",
+		"the namespace to fetch workloadentries from")
+	vmBSCommand.PersistentFlags().StringVarP(&organization, "organization", "o", "",
+		"the organization to use on the certificate, defaults to the same as the root cert.")
+	vmBSCommand.PersistentFlags().StringVar(&remoteDirectory, "remote-directory", "/var/run/istio",
+		"the directory to create on the remote machine.")
+	vmBSCommand.PersistentFlags().StringVar(&scpPath, "remote-scp-path", "/usr/bin/scp",
+		"the scp binary location on the target machine if not at /usr/bin/scp")
+	vmBSCommand.PersistentFlags().StringVar(&spiffeTrustDomain, "spiffee-trust-domain", "",
+		"The spiffee trust domain to set if not wanting to use the default.")
+	vmBSCommand.PersistentFlags().BoolVar(&sshIgnoreHostKeys, "ignore-host-keys", false,
+		"whether or not ot ignore host keys on the remote host")
+	vmBSCommand.PersistentFlags().StringVarP(&sshKeyLocation, "ssh-key", "k", "",
+		"the location of the ssh key to use to ssh into the machines")
```

Comment:
Capitalize SSH in the description and every message in this PR.  The "to use to ssh into the machines" could be eliminated.